src/dbnode/storage/index.go

```diff
@@ -1588,6 +1588,67 @@ func (i *nsIndex) CleanupDuplicateFileSets() error {
 	return multiErr.FinalError()
 }
 
+func (i *nsIndex) DebugMemorySegments(opts DebugMemorySegmentsOptions) error {
+	i.state.RLock()
+	defer i.state.RLock()
+	if i.state.closed {
+		return errDbIndexAlreadyClosed
+	}
+
+	ctx := context.NewContext()
+	defer ctx.Close()
+
+	// Create a new set of file system options to output to new directory.
+	fsOpts := i.opts.CommitLogOptions().
+		FilesystemOptions().
+		SetFilePathPrefix(opts.OutputDirectory)
+
+	for _, block := range i.state.blocksByTime {
+		segmentsData, err := block.MemorySegmentsData(ctx)
+		if err != nil {
+			return err
+		}
+
+		for numSegment, segmentData := range segmentsData {
+			indexWriter, err := fs.NewIndexWriter(fsOpts)
```

Comment:
Do we care to reuse this across segments?

Reply:
They're not reuseable at this time, heh.